File: SeriesGuide/src/preview/java/com/battlelancer/seriesguide/billing/amazon/AmazonIapManager.java
package com.battlelancer.seriesguide.billing.amazon;

import android.app.Activity;
import android.content.Context;

/**
 * No-op dummy of Amazon IAP manager.
 */
public class AmazonIapManager {

    public static void setup(Context context) {
        // no op
    }

    public static AmazonIapManager get() {
        return new AmazonIapManager();
    }

    public void requestUserDataAndPurchaseUpdates() {
        // no op
    }

    public void activate() {
        // no op
    }

    public void deactivate() {
        // no op
    }

    public void validateSubscription(Activity activity) {
        // no op
    }
}

File: app/src/pure/java/com/battlelancer/seriesguide/billing/amazon/AmazonIapManager.kt
package com.battlelancer.seriesguide.billing.amazon

import android.app.Activity
import android.content.Context

/**
 * No-op dummy of Amazon IAP manager.
 */
class AmazonIapManager(@Suppress("UNUSED_PARAMETER") context: Context) : AmazonIapManagerInterface {

    override fun register() {
        // no op
    }

    override fun requestProductData() {
        // no op
    }

    override fun requestUserDataAndPurchaseUpdates() {
        // no op
    }

    override fun activate() {
        // no op
    }

    override fun deactivate() {
        // no op
    }

    override fun validateSupporterState(activity: Activity) {
        // no op
    }

}
